### PR TITLE
COLLADA2GLTF 2.0

### PR DESCRIPTION
**Pull request by lasalvavida**
For #3

Current master has been checked out into the `1.0` branch. This PR would replace master with the `2.0` branch.

@emackey @pjcozzi

### CODE REVIEW

**Comment by pjcozzi** [comment]
Great news, @lasalvavida!

@lilleyse can you please test this out?

**Comment by lexaknyazev** [comment]
See https://github.com/KhronosGroup/glTF-Sample-Models/issues/59.

**Comment by lasalvavida** [comment]
This is updated from the feedback in [KhronosGroup/glTF-Sample-Models#59](https://github.com/KhronosGroup/glTF-Sample-Models/issues/59)

**Comment by lasalvavida** [comment]
PR for sample models is [KhronosGroup/glTF-Sample-Models#61](https://github.com/KhronosGroup/glTF-Sample-Models/pull/61)

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,12 @@
-[submodule "dependencies/OpenCOLLADA"]
-	path = dependencies/OpenCOLLADA
-	url = https://github.com/KhronosGroup/OpenCOLLADA.git
\ No newline at end of file
+[submodule "dependencies/OpenCOLLADA/OpenCOLLADA"]
```
Comment (emackey) [inline .gitmodules]: Moving this down a level makes it very hard to switch back and forth between 1.0 and 2.0.  Maybe the 1.0 one can also be moved down a level, to match?

Comment (lasalvavida) [inline .gitmodules]: That seems like a reasonable solution

**Comment by RemiArnaud** [comment]
My mac decided to update my compiler, now Xcode Version 8.2.1 

It wont compile this branch

first issue is the following line in the CMakeFile.txt
 
add_compile_options(-std=c++11 -lstdc++fs)

which affect 'C' and 'C++' files and break the compilation.

This can be replaced with set(CMAKE_CXX_STANDARD 11) as proposed by https://github.com/KhronosGroup/COLLADA2GLTF/issues/55

but then the compilation breaks with 
/Users/remi/git-mac/COLLADA2GLTF/GLTF/src/GLTFAccessor.cpp:5:10: 'experimental/filesystem' file not found

**Comment by emackey** [comment]
@lasalvavida What will it take to bring this branch into master?  2.0 is gaining a lot of traction and I think more people are getting confused that master still produces 1.0.  At some point we should just bring this into master and deal with remaining issues as separate issues.

**Comment by lasalvavida** [comment]
@emackey, see my notes on #83.

**Comment by donmccurdy** [comment]
Does that mean this PR is only blocked by not having an option to output glTF1.0? Is the `2.0` branch ready to use otherwise?

**Comment by pjcozzi** [comment]
> Does that mean this PR is only blocked by not having an option to output glTF1.0? Is the 2.0 branch ready to use otherwise?

@donmccurdy - @lilleyse is investigating what is left for the 2.0 branch.  We hope to know tomorrow. :crossed_fingers:

**Comment by lasalvavida** [comment]
1.0 branch is updated to the current master. This is merging now and will replace master with 2.0.

**Comment by pjcozzi** [comment]
wow, congrats @lasalvavida!  The community will be really excited about this.

Do you have a list of what features and testing are still missing from the 2.0 branch, e.g., morph targets, etc?  We can try to get help.

**Comment by donmccurdy** [comment]
Great work!! 🎊

+1 for @pjcozzi's question, and:

* Has anything changed w/r/t compiling on macOS? If not, is it possible to post a binary for macOS?
* Are there future plans for https://cesiumjs.org/convertmodel.html ? Or an issue to track for it?

**Comment by lasalvavida** [comment]
> wow, congrats @lasalvavida! The community will be really excited about this.

Thanks @pjcozzi! Definitely an exciting moment that's been a while in the making.

> Do you have a list of what features and testing are still missing from the 2.0 branch, e.g., morph targets, etc? We can try to get help.

Yup, morph targets still need support. As far as I know that's the only major 2.0 feature that we should support converting from COLLADA unless there is a PBR extension for COLLADA that I'm not aware of.

> Has anything changed w/r/t compiling on macOS? If not, is it possible to post a binary for macOS?

I have [Ahoy](https://github.com/lasalvavida/ahoy) building for macOS in CI now, so I will be using that CI build as a template to add a Travis build for macOS here.

> Are there future plans for https://cesiumjs.org/convertmodel.html ? Or an issue to track for it?

This is probably a better question for @pjcozzi since I don't work for AGI. I'm not far from emscripten builds working and that will likely be my next major focus on this project. When that happens, we'll likely host the client side conversion code as a static page somewhere.

**Comment by donmccurdy** [comment]
I’ll ask around internally, I’m not sure if we’re doing much with COLLADA right now... I think we would get some uptake from the A-Frame community with a web-based converter. Guessing client-side would take some time to do, but discussing (https://github.com/donmccurdy/three-gltf-viewer/issues/72) whether there is a straightforward way to do server side conversion in a “pluggable” setup.

**Pull request merged**